src/main/java/org/minions/devfund/angela/battleship/Ocean.java

```diff
@@ -0,0 +1,192 @@
+            while (!located) {
+                int row = random.nextInt(OCEAN_SIZE);
+                int column = random.nextInt(OCEAN_SIZE);
+                boolean horizontal = random.nextBoolean();
+                if (ship.okToPlaceShipAt(row, column, horizontal, this)) {
+                    ship.placeShipAt(row, column, horizontal, this);
+                    located = true;
+                }
+            }
+        }
+    }
+
+    /**
+     * Verifies if position was occupied.
+     *
+     * @param row    int row position.
+     * @param column int column position.
+     * @return true if position was occupied.
+     */
+    boolean isOccupied(final int row, final int column) {
+        return ships[row][column].getLength() > 1;
+    }
+
+    /**
+     * Shots a position.
+     *
+     * @param row    int row position.
+     * @param column int column position.
+     * @return true if ship was shot.
+     */
+    boolean shootAt(final int row, final int column) {
+        shotsFired++;
+        if (ships[row][column].shootAt(row, column)) {
+            if (ships[row][column].isSunk()) {
+                shipsSunk++;
+            }
+            hitCount++;
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Obtains the shots fired.
+     *
+     * @return int shots fired quantity.
+     */
+    int getShotsFired() {
+        return shotsFired;
+    }
+
+    /**
+     * Obtains the hit count.
+     *
+     * @return int he hit number.
+     */
+    int getHitCount() {
+        return hitCount;
+    }
+
+    /**
+     * Obtains the ships sunk quantity.
+     *
+     * @return int ships sunk number.
+     */
+    int getShipsSunk() {
+        return shipsSunk;
+    }
+
+    /**
+     * Verifies if is game over.
+     *
+     * @return true if game over.
+     */
+    boolean isGameOver() {
+        return shipsSunk == SHIPS_QUANTITY;
+    }
+
+    /**
+     * Gets the ship array.
+     *
+     * @return matrix of {@link Ship}
+     */
+    Ship[][] getShipArray() {
+        return ships;
+    }
+
+    /**
+     * Prints the ocean.
+     */
+    void print() {
+        StringBuilder sb = new StringBuilder();
+        sb.append(" ");
+        for (int i = 0; i < OCEAN_SIZE; i++) {
+            sb.append(String.format("%3d", i));
+        }
+        sb.append("\n");
+
+        for (int i = 0; i < OCEAN_SIZE; i++) {
+            sb.append(String.format("%2d ", i));
+            for (int j = 0; j < OCEAN_SIZE; j++) {
+                if (!ships[i][j].wasShootAt(i, j)) {
+                    if (!ships[i][j].getShipType().equals("empty")) {
+                        sb.append("a");
+                    } else {
+                        sb.append(".");
+                    }
+                } else {
+                    sb.append(ships[i][j].toString());
+                }
+                sb.append("  ");
+            }
+            sb.append("\n");
+        }
+        System.out.println(sb.toString());
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)